services/sharedchannel/sync_recv.go

```diff
@@ -106,7 +105,9 @@ func (scs *Service) processSyncMessages(syncMessages []syncMsg, rc *model.Remote
 			}
 
 			// process perma-links for remote
-			sm.Post.Message = scs.processPermalinkFromRemote(sm.Post, team)
+			if team != nil {
+				sm.Post.Message = scs.processPermalinkFromRemote(sm.Post, team)
```

Comment:
This means no perma-links will get corrected in DMs.  I don't think Product will go for that.  It's more work and probably out of scope for this PR, but we could modify `processPermalinkFromRemote` to look up the post via post id in the link and find out what the local team name is.  If you think that is out of scope, which is perfectly fair, then either of us can create a ticket.

Reply:
Yeah, I've put it on my TODO list and was about to create a ticket for that, since as you said it can be done outside of the scope of this ticket.

Reply:
Ticket created https://mattermost.atlassian.net/browse/MM-32813